oga/backend/users/views.py

```diff
@@ -46,33 +72,35 @@ def questions(request):
         question.save()
         response_dict = {'id': question.id}
         return JsonResponse(response_dict, status=201)
+    else:
+        return HttpResponse(status=405)
 
+@check_request
 @csrf_exempt
 def sign_up(request):
     if request.method == 'POST':
-        try:
-            req_data = json.loads(request.body.decode())
-            username = req_data['username']
-            password = req_data['password']
-        except (KeyError, json.JSONDecodeError):
-            return HttpResponseBadRequest()
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
         new_user = User.objects.create_user(username=username, password=password)
         response_dict = {'id': new_user.id}
         return JsonResponse(response_dict, status=201)
+    else:
+        return HttpResponseNotAllowed(['POST'])
 
+@check_request
 @csrf_exempt
 def sign_in(request):
     if request.method == 'POST':
-        try:
-            req_data = json.loads(request.body.decode())
-            username = req_data['username']
-            password = req_data['password']
-        except (KeyError, json.JSONDecodeError):
-            return HttpResponseBadRequest()
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
         user = authenticate(username=username, password=password)
         if user is not None:
             login(request, user)
             response_dict = {'id': user.id}
-            return JsonResponse(response_dict, status=201)
+            return JsonResponse(response_dict, status=204)
         else:
             return JsonResponse({}, status=401)
+    else:
```

Comment:
This can be checked by django's 
```python
@require_http_methods(["GET", "PUT", "DELETE"])
```
decorator.
Please remove the ifs and use this.`

Reply:
Thanks. I left if condition for future implementation but removed 'else:'